op-e2e/op_geth_test.go

```diff
@@ -44,9 +45,17 @@ func TestMissingGasLimit(t *testing.T) {
 // TestTxGasSameAsBlockGasLimit tests that op-geth rejects transactions that attempt to use the full block gas limit.
 // The L1 Info deposit always takes gas so the effective gas limit is lower than the full block gas limit.
 func TestTxGasSameAsBlockGasLimit(t *testing.T) {
+	if config.ExternalL2Nodes != "" {
+		// Some clients, such as Erigon will not discard txes that exceed the
+		// block gas limit, and instead retains them to re-evaluate with the
+		// next block whose gas limit may have increased.  We should enable this
+		// tests for external clients which exhibit this behavior, but currently
+		// we are disabling it for all external clients.
+		t.Skip("external client may not discard txes which exceed the gas limit")
+	}
```

Comment:
I suspect that we're better off omitting this currently and then when running tests with erigon skip tests it can't handle with the `-skip` option.

Reply:
I suspect the key will be to put the external client code in charge of kicking of the test run rather than the `op-e2e` module itself.  So rather than running `cd op-e2e; go test . -- --externalL2 external_geth/shim` you'd do something more like `cd external_geth; make test-e2e` and then make could ensure the client is built and run `go test` with the appropriate tests skipped. I'm not a make expert but ideally it would let you supply other test args so you could run a single test easily with the `-run <test>` option. `make` becomes the canonical entry point and external clients then have maximal flexibility to configure how the tests run, how the client builds, what to skip etc.